src/Comment.jsx

```diff
@@ -6,33 +6,47 @@ import { RIETextArea } from 'riek';
 import PropTypes from 'prop-types';
 import '../style/index.css';
 import TimeAgo from 'react-timeago';
+import CommentClient from './CommentClient';
 
-export default class Comment extends React.Component {
+if (!global.globalNameCache) {
+  global.globalNameCache = {};
+}
+
+export default class _Comment extends React.Component {
 
   constructor (props) {
     super(props);
+    this.commentClient = new CommentClient(props.accessToken, props.commentUri);
     this.state = {
-      comment: '',
-      createdBy: null,
-      createdByName: null,
-      createdAt: null,
-      updatedBy: null,
-      updatedByName: null,
-      updatedAt: null,
+      comment: (props.comment) ? props.comment.comment : '',
+      createdBy: (props.comment) ? props.comment.createdBy : '',
+      createdByName: (props.comment) ? global.globalNameCache[props.comment.createdBy] : null,
+      createdAt: (props.comment) ? props.comment.createdAt : '',
+      updatedBy: (props.comment) ? props.comment.updatedBy : '',
+      updatedByName: (props.comment) ? global.globalNameCache[props.comment.updatedBy] : null,
+      updatedAt: (props.comment) ? props.comment.updatedAt : '',
       visible: false,
-      ready: false
+      ready: props.comment != null
     };
   }
 
   componentWillReceiveProps (newProps) {
+    this.commentClient = new CommentClient(newProps.accessToken, newProps.commentUri);
     if (newProps.commentUri !== this.props.commentUri) {
       this.setState({
-        ready: false,
```

Comment:
If you send empty strings for values to the service it will use them to overwrite the existing values in the comment. I'm not sure that this is the expected behaviour. (I might also be missing some pruning step further down in the code.)

Reply:
[this.setState](https://reactjs.org/docs/react-component.html#setstate) updates the properties assigned and doesn't touch the ones the others. 

    updater is shallowly merged with prevState